frontend/src/features/idir/search/components/PermitResendDialog.tsx

```diff
@@ -1,127 +1,152 @@
+import { useEffect } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import { faEnvelope } from "@fortawesome/free-regular-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { FormLabel, Grid, OutlinedInput } from "@mui/material";
-import Button from "@mui/material/Button";
-import Dialog from "@mui/material/Dialog";
-import DialogActions from "@mui/material/DialogActions";
-import DialogContent from "@mui/material/DialogContent";
-import DialogTitle from "@mui/material/DialogTitle";
-import Typography from "@mui/material/Typography";
-import { useState } from "react";
-import { BC_COLOURS } from "../../../../themes/bcGovStyles";
+import { Button, Dialog } from "@mui/material";
+
+import "./PermitResendDialog.scss";
+import { Nullable } from "../../../../common/types/common";
+import { getDefaultRequiredVal } from "../../../../common/helpers/util";
+import { requiredMessage } from "../../../../common/helpers/validationMessages";
+import { CustomFormComponent } from "../../../../common/components/form/CustomFormComponents";
+
+interface PermitResendFormData {
+  permitId: string;
+  email: string;
+  fax?: Nullable<string>;
+}
+
+const FEATURE = "permit-resend";
 
 /**
  *  A dialog box for resending permit by email or fax.
  */
 export default function PermitResendDialog({
-  isOpen,
-  onClickResend,
-  onClickCancel,
+  shouldOpen,
+  onResend,
+  onCancel,
+  permitId,
+  permitNumber,
   email,
   fax,
-  permitNumber,
 }: Readonly<{
-  /**
-   * Boolean to control the open and close state of Dialog box.
-   */
-  isOpen: boolean;
-  /**
-   * A callback function on clicking resend button.
-   * @returns void
-   */
-  onClickResend?: () => void;
-
-  /**
-   * A callback function on clicking cancel button.
-   * @returns void
-   */
-  onClickCancel?: () => void;
-  /**
-   * The permit number to be displayed in the dialog box.
-   */
+  shouldOpen: boolean;
+  onResend: (
+    permitId: string,
+    email: string,
+    fax?: Nullable<string>,
+  ) => Promise<void>;
+  onCancel: () => void;
+  permitId: string;
   permitNumber: string;
-  /**
-   * The email address if available.
-   */
   email?: string;
-  /**
-   * The fax if available.
-   */
   fax?: string;
 }>) {
-  const [emailState, setEmailState] = useState<string>(email ?? "");
-  const [faxState, setFaxState] = useState<string>(fax ?? "");
+  const formMethods = useForm<PermitResendFormData>({
+    defaultValues: {
+      permitId,
+      email: getDefaultRequiredVal("", email),
+      fax: getDefaultRequiredVal("", fax),
+    },
+    reValidateMode: "onChange",
+  });
+
+  const { setValue, handleSubmit } = formMethods;
+
+  useEffect(() => {
+    setValue("permitId", permitId);
+  }, [permitId]);
+
+  useEffect(() => {
+    setValue("email", getDefaultRequiredVal("", email));
+  }, [email]);
+
+  useEffect(() => {
+    setValue("fax", getDefaultRequiredVal("", fax));
```

Comment:
Do we need these `useEffects`? just the form component with the correct name should take care of this isn't it?

Reply:
I was trying to think of what to do if the props for email/fax changed (by parent component) without changing values in the actual form (in which case the useEffect would update the form values), but I guess for this feature this isn't really a problem. I'll remove it.